next/src/hooks/useFullscreen.ts

```
@@ -0,0 +1,117 @@
+  isFullscreen: boolean;
+  enterFullscreen: () => void;
+  exitFullscreen: () => void;
+  toggleFullscreen: () => void;
+};
+
+const getDoc = () => (typeof window !== "undefined" ? window.document : undefined);
+
+export function useFullscreen(targetRef?: React.RefObject<Element>): FullscreenHook {
+  const [isFullscreen, setIsFullscreen] = useState(false);
+  // Used to keep the callback up-to-date
+  const ref = targetRef || useRef<Element>(getDoc()?.documentElement || null);
```

Biome (2.1.2)

[error] This hook is being called conditionally, but all hooks must be called in the exact same order in every component render.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)